t/t1300-config.sh

```diff
@@ -1766,6 +1766,87 @@ test_expect_success !MINGW '--show-origin blob ref' '
 	test_cmp expect output
 '
 
+
+test_expect_success '--show-scope with --list' '
+	cat >expect <<-EOF &&
+		(global)	user.global=true
+		(global)	user.override=global
+		(global)	include.path=$INCLUDE_DIR/absolute.include
+		(global)	user.absolute=include
+		(local)	user.local=true
+		(local)	user.override=local
+		(local)	include.path=../include/relative.include
+		(local)	user.relative=include
+		(command line)	user.cmdline=true
+	EOF
+	git -c user.cmdline=true config --list --show-scope >output &&
+	test_cmp expect output
+'
+
+test_expect_success '--show-scope with --list --null' '
+	cat >expect <<-EOF &&
+		(global)Quser.global
+		trueQ(global)Quser.override
+		globalQ(global)Qinclude.path
+		$INCLUDE_DIR/absolute.includeQ(global)Quser.absolute
+		includeQ(local)Quser.local
+		trueQ(local)Quser.override
+		localQ(local)Qinclude.path
+		../include/relative.includeQ(local)Quser.relative
+		includeQ(command line)Quser.cmdline
+		trueQ
+	EOF
+	git -c user.cmdline=true config --null --list --show-scope >output.raw &&
+	nul_to_q <output.raw >output &&
+	# The here-doc above adds a newline that the --null output would not
+	# include. Add it here to make the two comparable.
+	echo >>output &&
+	test_cmp expect output
+'
+
+test_expect_success '--show-scope with single file' '
+	cat >expect <<-\EOF &&
+		(local)	user.local=true
+		(local)	user.override=local
+		(local)	include.path=../include/relative.include
+	EOF
+	git config --local --list --show-scope >output &&
+	test_cmp expect output
+'
+
+test_expect_success '--show-scope with --get-regexp' '
+	cat >expect <<-EOF &&
+		(global)	user.global true
+		(local)	user.local true
+	EOF
+	git config --show-scope --get-regexp "user\.[g|l].*" >output &&
+	test_cmp expect output
+'
+
+test_expect_success '--show-scope getting a single key' '
+	cat >expect <<-\EOF &&
+		(local)	local
+	EOF
+	git config --show-scope user.override >output &&
+	test_cmp expect output
+'
```

Comment:
Correct me if I'm wrong, but don't all these tests exercise the precise same code path? So they would all be broken, or none of them, right?

Adding test cases is not necessarily as cheap a thing as one might think: it costs a lot of time in the Azure Pipelines _just_ because of all the Unix shell overhead (which is _particularly_ expensive on Windows).

I'd much prefer one test case that exercises all of these:

- a system setting
- a setting in a file included from `--system`
- a setting in `--global`
- a local setting
- a local setting overriding the `--global` setting

Reply:
I wasn't really sure on what the standard for testing was, so I just covered the same bases as show-origin... As far as your cases go:

- None of the other tests actually touch the system configuration file or option.  I believe this has to do with the git sdk not having /usr/etc/gitconfig file, and i'm not sure if the tests should be creating one or not.
- I'd also add an option passed in through `-c`, `--file`, or `--blob`
- I'd also want a test explicitly for the `--local` (or global or system) options considering the weird behavior with how the parser considers them not being of their respective scopes when they clearly are
- Interaction with `--show-origin` and `--null` probably don't need to be as tested but i'm not really confident enough to say one way or the other.  The other tests can probably be removed though